src/Symfony/Component/TypeInfo/Tests/TypeResolver/StringTypeResolverTest.php

```diff
@@ -76,6 +76,7 @@ public static function resolveDataProvider(): iterable
         // array shape
         yield [Type::arrayShape(['foo' => Type::true(), 1 => Type::false()]), 'array{foo: true, 1: false}'];
         yield [Type::arrayShape(['foo' => ['type' => Type::bool(), 'optional' => true]]), 'array{foo?: bool}'];
+        yield [Type::arrayShape(['foo' => Type::bool()], sealed: false), 'array{foo: bool, ...}'];
```

Comment:
the syntax for unsealed shapes actually also supports defining an array type to cover the allowed types for the extra keys.

Reply:
Oh, didn't know about that. For the record, this corresponds to this syntax: `array{foo: int, ...<string, bool>}`